src/Shared/FileUtilities.cs

```diff
@@ -1031,27 +1031,80 @@ internal static string MakeRelative(string basePath, string path)
             ErrorUtilities.VerifyThrowArgumentNull(basePath, nameof(basePath));
             ErrorUtilities.VerifyThrowArgumentLength(path, nameof(path));
 
-            if (basePath.Length == 0)
-            {
-                return path;
-            }
+            string fullBase = Path.GetFullPath(basePath);
+            string fullPath = Path.GetFullPath(path);
 
-            Uri baseUri = new Uri(EnsureTrailingSlash(basePath), UriKind.Absolute); // May throw UriFormatException
+            string[] splitBase = fullBase.Split(Path.DirectorySeparatorChar);
+            string[] splitPath = fullPath.Split(Path.DirectorySeparatorChar);
 
-            Uri pathUri = CreateUriFromPath(path);
-
-            if (!pathUri.IsAbsoluteUri)
+            if (!path.StartsWith(splitPath[0]))
```

Comment:
This seems suspicious to me, but I have no evidence off the top of my head 🤔

Reply:
Seeing if `fullPath` starts with the current working directory could be another way of determining this, though the current method is faster.

Reply:
One possible suspicious reason is the (bug) I just found for mac/linux: the path starts with `/`, which means this will always return false. Now fixed.